src/pages/dashboard/index.js

```diff
@@ -62,11 +64,11 @@ const Dashboard = () => {
 
   useEffect(() => {
     getAllPosts()
-    getAllUsers()
+    loggedInStudent && getMyCohort(loggedInStudent.cohortId)
```

Comment:
Would it be possible for `loggedInStudent.cohortId` to be undefined?

Reply:
I think it would then ultimately result in a GET to `cohorts/undefined/students`; would be better to handle this case if we know it happens

Reply:
It looks like the error I'm getting is actually to do with trying to access a property (cohortId) on something that is null (loggedInStudent). I think I misread the situation initially.

This is the error: `index.js:68 - Uncaught TypeError: Cannot read properties of null (reading 'cohortId')`

That said, I guess there still is a chance for loggedInStudent.cohortId to be undefined so I think I should add a .catch() somewhere to handle this if it does occur either way. 

Long story short, I guess there are two cases I need to handle here: 

- one where loggedInStudent itself is null, and trying to access one of its non existent properties causes the error above
- one where the cohortId property exists, but just happens to be null or undefined. 

Would this make sense?

Reply:
Update - I had a go at handling both. Not sure it's a very good solution, since it result in two 400 errors being fired systematically (:4000/cohorts/undefined/students:1 - Failed to load resource: the server responded with a status of 400 (Bad Request))

Reply:
My suggestion would be to detect if the needed value is not there and avoid making any call at all in those cases.
If you want, you can safely access `loggedInStudent.cohortId` without an error if `loggedInStudent` is undefined by using [optional chainig](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)

Reply:
(we can have a chat on Discord if you want, might be a bit complex to explain in detail otherwise)